### PR TITLE
Resolve HTTP tests warnings

### PR DESCRIPTION
**Pull request by aggarw13**
Remove (ARM) GCC warnings from HTTP tests

Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c [hunk 1]
```diff
@@ -324,31 +325,31 @@ TEST_TEAR_DOWN( HTTPS_Client_Unit_API )
...
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, ReadResponseStatusInvalidParameters );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, ReadResponseStatusSuccess );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, WriteRequestBodyInvalidParameters );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, WriteRequestBodySuccess );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, WriteRequestBodyNetworkSendFailure );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, ReadResponseBodyInvalidParameters );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, ReadResponseBodyNetworkReceiveFailure );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, ReadResponseBodyParsingFailure );
-    RUN_TEST_CASE( HTTPS_Client_Unit_API, ReadResponseBodySuccess );
+	RUN_TEST_CASE( HTTPS_Client_Unit_API, ConnectInvalidParameters );
```
Comment (sarenameas) [inline libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:328]: I don't think we want tabs here?

File: libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c [hunk 2]
```diff
@@ -358,89 +359,89 @@ TEST_GROUP_RUNNER( HTTPS_Client_Unit_API )
...
-
-    /* NULL certificates will result in a network error in the underlying network stack, but is allowed in the API so '
-     * that is not tested here. */
-
-    /* IotHttpsConnectionInfo_t.pNetworkInterface is NULL. */
-    testConnInfo.pNetworkInterface = NULL;
-    returnCode = IotHttpsClient_Connect( &connHandle, &testConnInfo );
-    TEST_ASSERT_NULL( connHandle );
-    TEST_ASSERT_EQUAL( IOT_HTTPS_INVALID_PARAMETER, returnCode );
+IotHttpsReturnCode_t returnCode = IOT_HTTPS_OK;
```
Comment (sarenameas) [inline libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:362]: This is incrustification is different and weird, but I mean if it passes the CI then ok... lol

File: libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c (hunk 1 above)
Comment (aggarw13) [reply libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c]: It was an accidental formatting change with the `libraries/freertos_plus/standard/freertos_plus_tcp/uncrustify.cfg` file

Comment (aggarw13) [reply libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c]: Reverted the formatting changes

File: libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c (hunk 2 above)
Comment (aggarw13) [reply libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c]: Accidental formatting changes from the FreeRTOS+TCP `uncrustify.cfg` file. Reverted

File: libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c [hunk 3]
```diff
@@ -798,7 +799,7 @@ TEST( HTTPS_Client_Unit_API, AddHeaderInvalidParameters )
     TEST_ASSERT_EQUAL( IOT_HTTPS_INVALID_PARAMETER, returnCode );
     returnCode = IotHttpsClient_AddHeader( reqHandle, HTTPS_HOST_HEADER, FAST_MACRO_STRLEN( HTTPS_HOST_HEADER ), HTTPS_TEST_ADDRESS, FAST_MACRO_STRLEN( HTTPS_TEST_ADDRESS ) );
     TEST_ASSERT_EQUAL( IOT_HTTPS_INVALID_PARAMETER, returnCode );
-    returnCode = IotHttpsClient_AddHeader( reqHandle, HTTPS_CONTENT_LENGTH_HEADER, FAST_MACRO_STRLEN( HTTPS_CONTENT_LENGTH_HEADER ), pTestContentLengthValueStr, FAST_MACRO_STRLEN( pTestContentLengthValueStr ) );
+    returnCode = IotHttpsClient_AddHeader( reqHandle, HTTPS_CONTENT_LENGTH_HEADER, FAST_MACRO_STRLEN( HTTPS_CONTENT_LENGTH_HEADER ), pTestContentLengthValueStr, strlen( pTestContentLengthValueStr ) );
```
Comment (aggarg) [inline libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:802]: Why this change?

Comment (aggarw13) [reply libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:802]: This was the warning associated with using the macro on a `char *`:
```
      |                                            ~~~~~~~~~^~~~~~
In file included from /home/ubuntu/Repos/amazon-freertos/libraries/c_sdk/standard/https/test/unit/iot_tests_https_common.h:43,
                 from /home/ubuntu/Repos/amazon-freertos/libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:31:
/home/ubuntu/Repos/amazon-freertos/libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c: In function 'TEST_HTTPS_Client_Unit_API_AddHeaderInvalidParameters_':
/home/ubuntu/Repos/amazon-freertos/libraries/c_sdk/standard/https/src/private/iot_https_internal.h:256:51: warning: division 'sizeof (char *) / sizeof (char)' does not compute the number of array elements [-Wsizeof-pointer-div]
  256 | #define FAST_MACRO_STRLEN( x )    ( ( sizeof( x ) / sizeof( char ) ) - 1 )
      |                                                   ^
/home/ubuntu/Repos/amazon-freertos/libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:801:162: note: in expansion of macro 'FAST_MACRO_STRLEN'
  801 |     returnCode = IotHttpsClient_AddHeader( reqHandle, HTTPS_CONTENT_LENGTH_HEADER, FAST_MACRO_STRLEN( HTTPS_CONTENT_LENGTH_HEADER ), pTestContentLengthValueStr, FAST_MACRO_STRLEN( pTestContentLengthValueStr ) );
      |                                                                                                                                                                  ^~~~~~~~~~~~~~~~~
/home/ubuntu/Repos/amazon-freertos/libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:772:12: note: first 'sizeof' operand was declared here
  772 |     char * pTestContentLengthValueStr = "0";
```

Comment (aggarw13) [reply libraries/c_sdk/standard/https/test/unit/iot_tests_https_client.c:802]: Basically, the operation `pTestContentLengthValueStr ` is of `char *` and the compiler warning is against use of non-array type (`sizeof( char *)` operation) for determining size of array
